src/main/kotlin/app/revanced/patches/tiktok/interaction/downloads/patch/DownloadsPatch.kt

```diff
@@ -50,7 +60,62 @@ class DownloadsPatch : BytecodePatch(
                 return v0
             """
         )
+        //Change the download path patch
+        val method4 = DownloadPathParentFingerprint.result!!.mutableMethod
+        val implementation4 = method4.implementation
+        val instructions = implementation4!!.instructions
+        var targetOffset = -1
+        //Search for the target method called instruction offset.
+        instructions.forEachIndexed { index, instruction ->
+            if (instruction.opcode != Opcode.CONST_STRING) return@forEachIndexed
+            val reference = (instruction as ReferenceInstruction).reference as StringReference
+            if (reference.string != "video/mp4") return@forEachIndexed
+            val targetInstruction = instructions[index + 1] as ReferenceInstruction
+            if (targetInstruction.opcode != Opcode.INVOKE_STATIC) return@forEachIndexed
+            targetOffset = index + 1
+        }
+        if (targetOffset == -1) return PatchResultError("Can not find download path uri method.")
+        //Change videos' download path.
+        val downloadUriMethod = data
+            .toMethodWalker(DownloadPathParentFingerprint.result!!.method)
+            .nextMethod(targetOffset, true)
+            .getMethod() as MutableMethod
+        downloadUriMethod.implementation!!.instructions.forEachIndexed { index, instruction ->
+            if (instruction.opcode == Opcode.SGET_OBJECT) {
+                val overrideRegister = (instruction as OneRegisterInstruction).registerA
+                downloadUriMethod.replaceInstruction(
+                    index,
+                    """
+                        const-string v$overrideRegister, "${downloadPath!!}"
+                    """
+                )
+            }
+            if (instruction.opcode == Opcode.CONST_STRING) {
+                val string = ((instruction as ReferenceInstruction).reference as StringReference).string
+                if (string.contains("/Camera")) {
+                    val overrideRegister = (instruction as OneRegisterInstruction).registerA
+                    val overrideString = string.replace("/Camera", "")
+                    downloadUriMethod.replaceInstruction(
+                        index,
+                        """
+                            const-string v$overrideRegister, "$overrideString"
+                        """
+                    )
+                }
+            }
+        }
         return PatchResultSuccess()
     }
 
+    companion object : OptionsContainer() {
+        private var downloadPath: String? by option(
+            PatchOption.StringOption(
+                key = "downloadPath",
+                default = "DCIM/Tiktok",
+                title = "Download Path",
+                description = "The custom download path. Allow directories are DCIM, Movies, Pictures.",
```

Comment:
Use a `StringListOption` instead if you're only allowing these values.

Reply:
How about the sub directory name? Should I create two options. One is for the parent directory (DCIM, Movies, Pictures). And another for sub directory like Tiktok,.... The restriction only apply to the parent directory due to the MediaStore limit.